.github/workflows/build.yaml

```diff
@@ -0,0 +1,138 @@
+name: build
+
+on:
+  push: {}
+  workflow_dispatch:
+      inputs:
+        version:
+          description: "Version to build, e.g. 0.1.0"
+          type: string
+          required: false
+
+env:
+  PKG_NAME: "vault-plugin-auth-azure"
+
+jobs:
+  get-go-version:
+    name: "Determine Go toolchain version"
+    runs-on: ubuntu-latest
+    outputs:
+      go-version: ${{ steps.get-go-version.outputs.go-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Determine Go version
+        id: get-go-version
+        run: |
+          echo "Building with Go $(cat .go-version)"
+          echo "::set-output name=go-version::$(cat .go-version)"
+
+  get-product-version:
+    runs-on: ubuntu-latest
+    outputs:
+      product-version: ${{ steps.get-product-version.outputs.product-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: get product version
+        id: get-product-version
+        run: |
+          VERSION="${{ github.event.inputs.version || '0.0.0-dev' }}"
+          echo "Using version ${VERSION}"
+          echo "::set-output name=product-version::${VERSION}"
+
+  generate-metadata-file:
+    needs: get-product-version
+    runs-on: ubuntu-latest
+    outputs:
+      filepath: ${{ steps.generate-metadata-file.outputs.filepath }}
+    steps:
+      - name: 'Checkout directory'
+        uses: actions/checkout@v2
+      - name: Generate metadata file
+        id: generate-metadata-file
+        uses: hashicorp/actions-generate-metadata@v1
+        with:
+          version: ${{ needs.get-product-version.outputs.product-version }}
+          product: ${{ env.PKG_NAME }}
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: metadata.json
+          path: ${{ steps.generate-metadata-file.outputs.filepath }}
+
+  test:
+    needs:
+      - get-go-version
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ needs.get-go-version.outputs.go-version }}
+      - name: Test
+        run: make bootstrap dev test
+
+  build:
+    needs:
+      - get-product-version
+      - get-go-version
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        arch: ["386", "arm", "arm64", "amd64"]
+        os: ["linux", "darwin", "windows", "freebsd", "openbsd", "netbsd", "solaris"]
+        exclude:
+          - os: darwin
+            arch: "386"
+          - os: darwin
+            arch: arm
+          - os: freebsd
+            arch: arm64
+          - os: netbsd # I guess "Of course it runs NetBSD" is no longer true
+            arch: arm64
+          - os: openbsd
+            arch: arm
```

Comment:
Looks like vault is building this combo, so we might want to include it for the plugins? https://github.com/hashicorp/vault/actions/runs/2586375558

Reply:
Hmm. I tried to add it and got:

```
No valid platforms to build for. If you specified a value
for the 'os', 'arch', or 'osarch' flags, make sure you're
using a valid value.
```

Let me see how Vault main gets a build of openbsd arm.

Reply:
Yeah from what I recall, CRT builds don't use/need gox since each os/arch combo is built in a separate job.

Reply:
We could switch to using `GOOS`, etc., if I don't get any traction about updating `gox`

We still use separate builds for performance (it can take several minutes for 1 job to build every platform we require).